lib/datadog/tracing/utils.rb

```diff
@@ -43,6 +43,34 @@ def self.reset!
       end
 
       private_class_method :id_rng, :reset!
+
+      # The module handles bitwise operation for trace id
+      module TraceId
+        MAX = (1 << 128) - 1
+
+        module_function
+
+        def next_id
+          return Utils.next_id unless Datadog.configuration.tracing.trace_id_128_bit_generation_enabled
+
+          concatenate(
+            Time.now.to_i << 32,
+            Utils.next_id
+          )
+        end
+
+        def to_high_order(trace_id)
+          trace_id.to_i >> 64
+        end
+
+        def to_low_order(trace_id)
+          trace_id.to_i & 0xFFFFFFFFFFFFFFFF
+        end
+
+        def concatenate(high_order, low_order)
+          high_order.to_i << 64 | low_order.to_i
```

Comment:
Let's remove `#to_i` calls from here, given that we only expect `Integer`s or `nil` as arguments.
`#to_i` is a very lenient method that easily allows for unintended objects to be converted to numbers, and that's not the indented use for these methods.